### ISSUE TITLE
Better File Structure

### ISSUE THREAD

**Issue by tdd11235813** [mpicbg-scicomp/gearshifft#31]
At the moment inc/ is just flat. I would suggest to structure the code file layout as following:

```
./inc/
__./core/
.......<gearshifft_core_headers>
__./libraries/
____./cufft/
.......cufft.hpp
.......cufft_helper.hpp
____./clfft/
____./fftw/
```

Names should be all lowercase. `./src/` could be kept flat due to the few files. Probably `benchmark.cpp` can be renamed to gearshifft.cpp.
I also have another benchmark in my mind (callbacks), maybe there will be a second benchmark path within gearshifft and the abstract fft class gets a second measurement layout (for the callbacks), where something like `gearshifft_callback.cpp` is an option for the second path.
Just brainstorming...

**Comment by psteinb** [comment]
I'd agree to that, although I'd consider something along the lines of:

```
./inc/
__./core/
.......<gearshifft_core_headers>
__./libraries/
.......cufft.hpp
.......cufft_helper.hpp
```

to be sufficient. In the end, it's merely 2-3 headers per new library 
binding. Which I'd say is manageable. But feel free to do as you 
proposed, if you think it helps with regard to future expansions.

**Comment by tdd11235813** [comment]
closed and resolved with #35
